### ISSUE TITLE
Twig in content: isSafe() really called ?

### ISSUE THREAD

**Issue by jmschelcher** [bolt/bolt#2487]
Hi,

I have a problem with the isSafe() function that permits to use Twig functions in Content and not only in templates.
I understood that this function must be added in the Extension.php of the extension (tried using simpleform extension here: extensions/vendor/bolt/simpleforms/Extension.php).

These extension are loaded in src/Extensions.php at line ~291.
I think that we call the is_callable and isSafe on the bad $extension (because the variable name $extension is use for 2 different objects:

The original is:
```php
        // Add Twig extensions
        if (is_callable(array($extension, 'getTwigExtensions'))) {
            /** @var \Twig_Extension[] $extensions */
            $extensions = $extension->getTwigExtensions();
            $addTwigExFunc = array($this, 'addTwigExtension');
            foreach ($extensions as $extension) {
                $this->app['twig'] = $this->app->share(
                    $this->app->extend(
                        'twig',
                        function(\Twig_Environment $twig) use ($addTwigExFunc, $extension, $name) {
                            call_user_func($addTwigExFunc, $twig, $extension, $name);
                            return $twig;
                        }
                    )
                );

                if (is_callable(array($extension, 'isSafe')) && $extension->isSafe() === true) {
                    $this->app['safe_twig'] = $this->app->share(
                        $this->app->extend(
                            'safe_twig',
                            function(\Twig_Environment $twig) use ($addTwigExFunc, $extension, $name) {
                                call_user_func($addTwigExFunc, $twig, $extension, $name);
                                return $twig;
                            }
                        )
                    );
                }
            }
```

I propose to change into:
(just renamed _some_ $extensions to $twig_extensions)

```php
          // Add Twig extensions
          if (is_callable(array($extension, 'getTwigExtensions'))) {
              /** @var \Twig_Extension[] $extensions */
              $twig_extensions = $extension->getTwigExtensions();
              $addTwigExFunc = array($this, 'addTwigExtension');
              foreach ($twig_extensions as $twig_extension) {
                  $this->app['twig'] = $this->app->share(
                      $this->app->extend(
                          'twig',
                          function(\Twig_Environment $twig) use ($addTwigExFunc, $twig_extension, $name) {
                              call_user_func($addTwigExFunc, $twig, $twig_extension, $name);
                              return $twig;
                          }
                      )
                  );

                  if (is_callable(array($extension, 'isSafe')) && $extension->isSafe() === true) {
                      $this->app['safe_twig'] = $this->app->share(
                          $this->app->extend(
                              'safe_twig',
                              function(\Twig_Environment $twig) use ($addTwigExFunc, $twig_extension, $name) {
                                  call_user_func($addTwigExFunc, $twig, $twig_extension, $name);
                                  return $twig;
                              }
                          )
                      );
                  }
              }
```

And now, it works...

Am I right or did I put the isSafe() function in a wrong way ?!

Thanks

**Comment by bobdenotter** [comment]
Hi @jmschelcher, 

Good catch. It seems that when we did some refactoring, we overlooked this: `foreach ($extensions as $extension) {`, as it overwrites the present `$extension`. 

Do you want to make a Pull Request, for this, so you get the 'credits' for fixing it?

**Comment by jmschelcher** [comment]
huhu why not, i'll try that

**Comment by bobdenotter** [comment]
Closed by #2492.
